_test/general.test.php

```diff
@@ -35,27 +95,364 @@ public function test_plugininfo() {
      * Test to ensure that every conf['...'] entry in conf/default.php has a corresponding meta['...'] entry in
      * conf/metadata.php.
      */
-    public function test_plugin_conf() {
-        $conf_file = __DIR__.'/../conf/default.php';
-        if (file_exists($conf_file)){
+    public function test_plugin_conf()
+    {
+        $conf_file = __DIR__ . '/../conf/default.php';
+        if (file_exists($conf_file)) {
             include($conf_file);
         }
-        $meta_file = __DIR__.'/../conf/metadata.php';
+        $meta_file = __DIR__ . '/../conf/metadata.php';
         if (file_exists($meta_file)) {
             include($meta_file);
         }
 
-        $this->assertEquals(gettype($conf), gettype($meta),'Both ' . DOKU_PLUGIN . 'findologicxmlexport/conf/default.php and ' . DOKU_PLUGIN . 'findologicxmlexport/conf/metadata.php have to exist and contain the same keys.');
+        $this->assertEquals(gettype($conf), gettype($meta), 'Both ' . DOKU_PLUGIN . 'findologicxmlexport/conf/default.php and ' . DOKU_PLUGIN . 'findologicxmlexport/conf/metadata.php have to exist and contain the same keys.');
 
         if (gettype($conf) != 'NULL' && gettype($meta) != 'NULL') {
-            foreach($conf as $key => $value) {
+            foreach ($conf as $key => $value) {
                 $this->assertArrayHasKey($key, $meta, 'Key $meta[\'' . $key . '\'] missing in ' . DOKU_PLUGIN . 'findologicxmlexport/conf/metadata.php');
             }
 
-            foreach($meta as $key => $value) {
+            foreach ($meta as $key => $value) {
                 $this->assertArrayHasKey($key, $conf, 'Key $conf[\'' . $key . '\'] missing in ' . DOKU_PLUGIN . 'findologicxmlexport/conf/default.php');
             }
         }
 
     }
+
+    //======================================================================
+    // XML TESTS
+    //======================================================================
+
+    /**
+     * Test to ensure that XML is valid when DokuWiki has no pages.
+     */
+    public function test_xml_response_is_valid_if_dokuwiki_has_no_pages()
```

Comment:
This, and the following similar tests, also look like good candidates for a data provider.

Reply:
Yes thanks for this.